lib/mayaUsd/ufe/UsdUndoCreateStageWithNewLayerCommand.h

```diff
@@ -60,12 +60,14 @@ class MAYAUSD_CORE_PUBLIC UsdUndoCreateStageWithNewLayerCommand
     void redo() override;
 
 private:
+    // Executes the command, called within a undo recorder.
+    // Returns true on success.
+    bool executeWithinUndoRecorder();
+
     Ufe::SceneItem::Ptr _parentItem;
     Ufe::SceneItem::Ptr _insertedChild;
 
-    MDagModifier& _createTransformDagMod;
-    MDagModifier& _createProxyShapeDagMod;
-    bool          _success;
+    OpUndoItemList _undoItemList;
 }; // UsdUndoCreateStageWithNewLayerCommand
```

Comment:
Since you're reviewing this, take note that anything that is related to the OpUndoItem need to have somewhere at teh root of the code an OpUndoItemList and OpUndoItemRecorder or OpUndoItemMuting. I have noted to try to improve them by adding debug-only code to verify that undo item are not left orphaned. The API having a global hidden state makes it easy to misuse accidentally.

Reply:
That is good to know, I was not aware of that. Thanks for pointing it out!